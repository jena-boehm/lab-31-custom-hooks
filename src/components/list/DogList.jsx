import React from 'react';
import PropTypes from 'prop-types';
import Dog from './Dog';
import Header from '../header/Header';
import styles from '../app/styles/App.css';

const DogList = ({ dogs }) => {
  const renderDogs = dogs.map(dog => (
    <li key={dog.name}>
      <Dog 
        name={dog.name}
        image={dog.image} />
    </li>
  ));

  return (
    <div className={styles.app}>
      <div className={styles.headerContainer}>
        <Header />
      </div>
      <div className={styles.body}>
        <div className={styles.list}>
          <ul data-testid="dogs"
            className={styles.ul}>
            {renderDogs}
          </ul>
        </div>
      </div>
    </div>
  );
};

DogList.propTypes = {
  dogs: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    image: PropTypes.string
  }))
};

export default DogList;
