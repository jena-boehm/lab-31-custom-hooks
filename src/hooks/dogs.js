import { useState, useEffect } from 'react';
import { getDogsByBreed } from '../services/DogApi';

export const useDogs = () => {
  const [loading, setLoading] = useState(true);
  const [dogs, setDogs] = useState([]);

  useEffect(() => {
    getDogsByBreed()
      .then(dogs => {
        setDogs(dogs);
        setLoading(false);
      });
  }, []);

  return {
    loading,
    dogs
  };
};
