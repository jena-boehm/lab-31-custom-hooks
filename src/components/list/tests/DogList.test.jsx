import React from 'react';
import { render, cleanup } from '@testing-library/react';
import DogList from '../DogList';
import { MemoryRouter } from 'react-router-dom';
import dogs from '../../../fixtures/dogs.json';

describe('DogList component', () => {
  afterEach(() => cleanup());
  it('renders DogList', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <DogList dogs={[dogs]} />
      </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
