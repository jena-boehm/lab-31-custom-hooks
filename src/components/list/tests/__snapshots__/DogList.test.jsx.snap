// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogList component renders DogList 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="headerContainer"
    >
      <div
        class="header"
      >
        <img
          alt="logo"
          class="logo"
          src="pawprint.png"
          width="70px"
        />
        <a
          class="headerLink"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="body"
    >
      <div
        class="list"
      >
        <ul
          class="ul"
          data-testid="dogs"
        >
          <li>
            <figure
              class="listItem"
            >
              <figcaption
                class="name"
              />
              <a
                href="/breeds/undefined"
              >
                <img
                  class="image"
                  height="200px"
                />
              </a>
            </figure>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
